example/plugin.h

```diff
@@ -0,0 +1,21 @@
+#include "proxy_wasm_intrinsics.h"
+
+class PluginRootContext : public RootContext {
+ public:
+  explicit PluginRootContext(uint32_t id, std::string_view root_id)
+      : RootContext(id, root_id) {}
+
+  bool onConfigure(size_t) override;
+};
+
+class PluginContext : public Context {
+ public:
+  explicit PluginContext(uint32_t id, RootContext* root) : Context(id, root) {}
+
```

Comment:
should we include all major functions that are available to override here?

Reply:
No, this is deemed to be super simple. Proxy sdk should document available callbacks, which is current missing.

Reply:
Created https://github.com/proxy-wasm/proxy-wasm-cpp-sdk/issues/78 to track this.